docs/elasticsearch/ux.md

```diff
@@ -0,0 +1,1240 @@
+1
+2
+3
+4
+5
+6
+7
+8
+9
+10
+------------- from: 10, size: 10
+11
+12
+13
+14
+15
+16
+17
+18
+19
+20
+```
+
+To calculate the `from` parameter relative to the page number:
+
+```json
+from = size * (page_number - 1)
+```
+
+Each time the user clicks the next page of the results, your application must make the same search query with an incremented `from` value.
+
+You could also specify the `from` and `size` parameters in the search URI:
+
+```json
+GET shakespeare/_search?from=0&size=10
+```
+
+If you only specify the `size` parameter, the `from` parameter defaults to 0.
+
+Querying for pages deep in your results could have a significant performance impact, so this approach is limited to 10000 results.
+
+The `from` and `size` parameters are completely stateless, so the results are always based on the latest available data.
+This could cause inconsistent pagination.
+For example, assume a user stays on the first page of the results for a minute and then navigates to the second page; in that time, a new document is indexed in the background which is relevant enough to show up on the first page. In this scenario, the last result of the first page is pushed to the second page, so the user ends up seeing a result on the second page that they already saw on the first page.
+
+Use the `scroll` operation for consistent pagination. With the `scroll` operation, a search context is kept open for a certain period of time. Any data changes do not affect the results during this time.
+
+## Scroll
+
+The `from` and `size` parameters allow you to paginate your search results, but with a limit of 10000 results at a time. If you need to request massive volumes of data for say a machine learning job, use the `scroll` operation instead. The `scroll` operation allows you to request an unlimited number of results.
+
+To use the scroll operation, add a `scroll` parameter to the request header with a search context to tell Elasticsearch how long you need to keep scrolling. This search context needs to be long enough to process a single batch of results. The `size` parameter allows you to set the number of results that you want returned for each batch.
+
+```json
+GET shakespeare/_search?scroll=10m
+{
+  "size": 10000
+}
+```
+
+Elasticsearch caches the results and returns a scroll ID to access them in batches:
+
+```json
+"_scroll_id" : "DXF1ZXJ5QW5kRmV0Y2gBAAAAAAAAAAUWdmpUZDhnRFBUcWFtV21nMmFwUGJEQQ=="
+```
+
+Pass this scroll ID to the `scroll` operation to get back the next batch of results:
+
+```json
+GET _search/scroll
+{
+  "scroll": "10m",
+  "scroll_id": "DXF1ZXJ5QW5kRmV0Y2gBAAAAAAAAAAUWdmpUZDhnRFBUcWFtV21nMmFwUGJEQQ=="
+}
+```
+
+Using this scroll ID, you get results in batches of 10000 as long as the search context is still open. Typically, the scroll ID does not change between requests. But it could change, so make sure you always use the latest scroll ID. If the next scroll request is not sent within the set search context, the results are not returned.
```

Comment:
does not change between requests, but it *can* change, so make sure to always use the latest scroll ID.

Reply:
Done.